src/rulesets/__tests__/validation.test.ts

```diff
@@ -153,4 +154,116 @@ describe('Ruleset Validation', () => {
       }),
     ).toThrow();
   });
+
+  it('recognizes functions directory', () => {
+    expect(
+      assertValidRuleset.bind(null, {
+        functionsDir: 'baz',
+        rules: {},
+      }),
+    ).not.toThrow();
+  });
+
+  it('recognizes invalid functions directory', () => {
+    expect(
+      assertValidRuleset.bind(null, {
+        functionsDir: 2,
+        rules: {},
+      }),
+    ).toThrow();
```

Comment:
Do we care what its throwing? Exception names can be considered part of the interface so if an exception type changes its actually a breaking change, and would mean we need to bump a major under SemVer. Should we check?

Reply:
These errors are schema errors reported by AJV, yet I can check for them so that we'll know that a new version of ajv changes something :+1:

Reply:
It'd look more or less like this
```ts
it('recognizes invalid schema functions', () => {
    expect(
      assertValidRuleset.bind(null, {
        functions: ['d', { typo: 'a' }],
        rules: {},
      }),
    ).toThrow(`/functions/1 should be string
/functions/1 should be array
/functions/1 should match exactly one schema in oneOf`);
  });
```

Reply:
Oh I meant the _type_ of error, so `AjvException` or whatever, if that's a thing in TS.